lib/Convos/Core/Backend/File.pm

```diff
@@ -80,14 +80,26 @@ sub messages {
   $re = qr{\Q$re\E}i unless ref $re;
   $re = qr/^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}) (.*$re.*)$/;
 
-  $args{after}  = Time::Piece->strptime($query->{after},  '%Y-%m-%dT%H:%M:%S') if $query->{after};
-  $args{before} = Time::Piece->strptime($query->{before}, '%Y-%m-%dT%H:%M:%S') if $query->{before};
-
-  # If neither before nor after are defined, search everything before now
-  $args{before} = gmtime if !$args{before} && !$args{after};
   $args{limit}    = $query->{limit} || 60;
   $args{messages} = [];
   $args{re}       = $re;
+  if ( !$query->{before} && !$query->{after} ) {
+    $args{before} = gmtime;
+    $args{after} = $args{before}->add_months(-12);
+  } elsif ( $query->{before} && !$query->{after} ) {
+    $args{before} = Time::Piece->strptime($query->{before}, '%Y-%m-%dT%H:%M:%S');
+    $args{after} = $args{before}->add_months(-12);
+  } elsif ( !$query->{before} && $query->{after} ) {
+    $args{after} = Time::Piece->strptime($query->{after}, '%Y-%m-%dT%H:%M:%S');
+    $args{before} = $args{after}->add_months(12);
+  } elsif ( $query->{before} && $query->{after} ) {
+    $args{before} = Time::Piece->strptime($query->{before}, '%Y-%m-%dT%H:%M:%S');
+    $args{after} = Time::Piece->strptime($query->{after}, '%Y-%m-%dT%H:%M:%S');
+    # return no messages if args > before or before - args > 12
+    $args{after} = $args{before}->add_months(1) if $args{before} - $args{after} > $args{before} - $args{before}->add_months(-12);
+  }
+  $args{cursor} = $args{before};
```

Comment:
> I think the if/else/elsif above could collapse to:

This misses an important case: if "after" is supplied but not "before" I want "before" to be "after" plus 12 months.  This series of conditionals allows for that situation which allows for a search to be limited to 12 months which is both less expensive and not prone to DoS.

Since you approve of my original code I'll leave it and comment it.  Indeed, improvements can always be made later!

I'll make my comments, and merge.  Thanks for all the feedback!